packages/extension-table/src/TableView.ts

```diff
@@ -1,49 +1,55 @@
-// @ts-nocheck
 import { Node as ProseMirrorNode } from '@tiptap/pm/model'
 import { NodeView } from '@tiptap/pm/view'
 
+import { getColStyleDeclaration } from './utilities/colStyle.js'
+
 export function updateColumns(
   node: ProseMirrorNode,
-  colgroup: Element,
-  table: Element,
+  colgroup: HTMLTableColElement, // <colgroup> has the same prototype as <col>
+  table: HTMLTableElement,
   cellMinWidth: number,
   overrideCol?: number,
-  overrideValue?: any,
+  overrideValue?: number,
 ) {
   let totalWidth = 0
   let fixedWidth = true
   let nextDOM = colgroup.firstChild
   const row = node.firstChild
 
-  for (let i = 0, col = 0; i < row.childCount; i += 1) {
-    const { colspan, colwidth } = row.child(i).attrs
-
-    for (let j = 0; j < colspan; j += 1, col += 1) {
-      const hasWidth = overrideCol === col ? overrideValue : colwidth && colwidth[j]
-      const cssWidth = hasWidth ? `${hasWidth}px` : ''
+  if (row !== null) {
+    for (let i = 0, col = 0; i < row.childCount; i += 1) {
+      const { colspan, colwidth } = row.child(i).attrs
 
-      totalWidth += hasWidth || cellMinWidth
+      for (let j = 0; j < colspan; j += 1, col += 1) {
+        const hasWidth = overrideCol === col ? overrideValue : (colwidth && colwidth[j]) as number | undefined
+        const cssWidth = hasWidth ? `${hasWidth}px` : ''
 
-      if (!hasWidth) {
-        fixedWidth = false
-      }
+        totalWidth += hasWidth || cellMinWidth
 
-      if (!nextDOM) {
-        colgroup.appendChild(document.createElement('col')).style.width = cssWidth
-      } else {
-        if (nextDOM.style.width !== cssWidth) {
-          nextDOM.style.width = cssWidth
+        if (!hasWidth) {
+          fixedWidth = false
         }
 
-        nextDOM = nextDOM.nextSibling
+        if (nextDOM === null) {
```

Comment:
Sorry for the nit, would prefer to keep this as
```suggestion
        if (!nextDOM) {
```
Just in case it is undefined, and to reduce differentiation from the prior behavior